test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,141 @@ func testManualMeteringInstall(
+		return nil
+	}
+	if apierrors.IsNotFound(err) {
+		n := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ctx.Namespace,
+				Labels: map[string]string{
+					"name": ctx.Namespace + "-" + testingNamespaceLabel,
+				},
+			},
+		}
+		_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), n, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		ctx.Logger.Debugf("Created the %s namespace", ctx.Namespace)
+	}
+
+	//decode and create yamls except for persistentvolume
+
+	type nfsManifestItem struct {
+		resourceType interface{}
+		resourceFile string
+		actions      []string
+	}
+	type nfsServer struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsService struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+
+	type nfsStorageclass struct {
+		CRD *apiextv1.CustomResourceDefinition
+	}
+	var nfsManifestItems = map[string]nfsManifestItem{
+		"nfsServer":
+		{
+
+			resourceType: nfsServer{},
+			resourceFile: "/test/testhelpers/nfs_server.yaml",
+			actions:      []string{"CREATE"},
+		},
+		"nfsService":
+		{
+			resourceType: nfsService{},
+			resourceFile: "/test/testhelpers/nfs_service.yaml",
+			actions:      []string{"CREATE", "GET"},
+		},
+		"nfsStorageclass":
+		{
+			resourceType: nfsStorageclass{},
+			resourceFile: "/test/testhelpers/nfs_storageclass.yaml",
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfsManifestItems[0].resourceType, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"],nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+
+				foo, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.Background(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
+
+	//  - Wait for service to populate spec.ClusterIP (poll.Wait functionality)
+	// poll has a closure that polls for the server IP to get populated
+	err := wait.Poll(3*time.Second, 5*time.Minute, func() (done bool, err error) {
+
+		pvc, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
```

Comment:
It looks like you're querying for the "nfs-service" Service in the `ctx.Namespace` namespace, but assigning to a variable called `pvc`.

Reply:
Also, you should query for this resource (`.CoreV1().Service(...).Get(...)`) before polling. That way, if the resource already exists and the spec.ClusterIP has already been populated, you don't need to poll and continue to ping the apiserver. If the spec.ClusterIP has not been populated yet, then you run the poll logic, overwriting the variable name you used for the first GET, which should solve the scoping problem of the poll conditional function.